tests/audit-tests/src/test/java/uk/gov/di/txma/audit/bdd/PassportCRI_TxmaStepDefinitions.java

```diff
@@ -0,0 +1,205 @@
+        BrowserUtils.waitForPageToLoad(10);
+        new IPVCoreFrontPage().UkPassport.click();
+        BrowserUtils.waitForPageToLoad(10);
+    }
+    @When("user completes address journey successfully")
+    public void fillInPassportDetails() throws JsonProcessingException {
+
+        WebElement PassportNumber = Driver.get().findElement(By.id("passportNumber"));
+        PassportNumber.sendKeys("824159121");
+        WebElement Surname = Driver.get().findElement(By.id("surname"));
+        Surname.sendKeys("Watson");
+        WebElement FirstName = Driver.get().findElement(By.id("firstName"));
+        FirstName.sendKeys("Mary");
+        WebElement birthDay = Driver.get().findElement(By.id("dateOfBirth-day"));
+        birthDay.sendKeys("25");
+        WebElement birthMonth = Driver.get().findElement(By.id("dateOfBirth-month"));
+        birthMonth.sendKeys("02");
+        WebElement birthYear = Driver.get().findElement(By.id("dateOfBirth-year"));
+        birthYear.sendKeys("1932");
+        WebElement PassportExpiryDay = Driver.get().findElement(By.id("expiryDate-day"));
+        PassportExpiryDay.sendKeys("01");
+        WebElement PassportExpiryMonth = Driver.get().findElement(By.id("expiryDate-month"));
+        PassportExpiryMonth.sendKeys("03");
+        WebElement PassportExpiryYear = Driver.get().findElement(By.id("expiryDate-year"));
+        PassportExpiryYear.sendKeys("2031");
+        WebElement PassportContinueButton = Driver.get().findElement(By.xpath("//button[@class='govuk-button button']"));
+        PassportContinueButton.click();
+        new VerifiableCredentialsPage().PassportCredential_attributes_link.click();
+        String PassportCRIJSONResponse = new VerifiableCredentialsPage().PassportCRIJSONResponse.getText();
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode jsonNode = objectMapper.readTree(PassportCRIJSONResponse);
+        sub = jsonNode.get("userId").asText();
+        Driver.get().quit();
+
+    }
+
+    @Then("the audit event should appear in TxMA")
+    public void checkPassportCriEventInTxMAS3()  throws InterruptedException {
+        assertTrue(isFoundInS3());
+
+    }
+
+    public boolean isFoundInS3() throws InterruptedException {
+        // count will make sure it only searches for a finite time
+        int count = 0;
+
+        // Has a retry loop in case it finds the wrong key on the first try
+        // Count < 11 is enough time for it to be processed by the Firehose
+        while (count < 11) {
+            // Checks for latest key and saves the contents in the output variable
+            output = new ArrayList<>();
+            findLatestKeysFromAuditS3();
+
+            // If an object was found
+            if (output != null) {
+
+                // Compares all individual jsons with our test data
+                for (JSONObject object : output) {
+                    if (object.toString().contains(sub)) {
```

Comment:
As we talked about briefly yesterday, this is actually potentially checking for the wrong event...

It actually checks for any event with the user_id (stored in `sub`) but the IPV Core events are also being sent with this stub. We should also include a passport event - IPV_PASSPORT_CRI_START - in the search as well. This can be made generic by adding it to the inputs to the function.